metricbeat/module/kafka/partition/partition.go

```diff
@@ -91,21 +97,17 @@ func (m *MetricSet) connect() (*sarama.Broker, error) {
 		return nil, err
 	}
 
-	addr := b.Addr()
-	for _, other := range meta.Brokers {
-		if other.Addr() == addr {
-			m.id = other.ID()
-			break
-		}
-	}
-
-	if m.id == noID {
-		closeBroker(b)
-		err = fmt.Errorf("No advertised broker with address %v found", addr)
-		return nil, err
+	other := findMatchingBroker(b.Addr(), meta.Brokers)
+	if other != nil {
+		debugf("found matching broker %v with id %v", other.Addr(), other.ID())
+		m.id = other.ID()
+		return b, nil
 	}
 
-	return b, nil
+	// no broker if found
+	closeBroker(b)
```

Comment:
Could we turn that around, that we have if other == nil we return the error and close the broker and move the non error case tot he bottom? Would make it easier readable.